### ISSUE TITLE
Consider not vendoring the C code

### ISSUE THREAD

**Issue by saxbophone** [ashleynewson/SmartSim#7]
Maybe you have a good reason for storing the C code in source control for this project, but as I understand it they're auto-generated, as they're what the Vala compiler outputs.

Standard practice normally is to not 'vendor' these files as they can be programatically re-constructed by running the Vala compiler (similar practice is seen for minified JavaScript, C/C++ object files and executables, etc...)

Also, if Vala introduce some new improvements to the way they build the C code, this project won't immediately benefit from it as the changes will have to be 'baked in'.

Be good to know what your thoughts are on the matter.

**Comment by ashleynewson** [comment]
Your statements are perfectly valid, and I did indeed consider them at the time. Maintaining the C code version takes effort, and is somewhat unclean and unconventional; however, I did have my reasons.

My original reason for vendoring the C code was to make it easier for people without the Vala compiler to compile SmartSim. Additionally, C is more future-proofed against potential compatibility-breaking updates to the Vala language (which I recall actually happened at some point). Of course, the C code doesn't need to be version controlled to the same level as the Vala code, so it doesn't need to be in the Git repo; however, this was, at the time (a number of years ago), more convenient. Things have since improved (e.g. Vala stability and GitHub features), and there are now better ways to organise things.

If I was not the sole developer, or if I made more frequent changes to SmartSim, I would have provided an alternate C-code snapshot separate from the main repository.

If I make any future modifications to SmartSim, I will reconsider this issue, and likely move the C code into a separate release package.
